swift/release/6.1: [X86] Disable shrink-wrapping for tailcc/swifttailcc functions.

ahmedbougacha opened the pull request:
Pseudo-expansion's handling of tail-calls currently interacts badly with shrink-wrapping when using callee-pop CCs on x86_64, resulting in an unbalanced RSP adjustment on shrink-wrapped return paths.

Disable shrink-wrapping until that's resolved, if it can be.

See https://github.com/llvm/llvm-project/issues/109279

rdar://141673907
(cherry picked from commit 14065737a74eba248122bd9f2541b23432221048)

ahmedbougacha committed:
[X86] Disable shrink-wrapping for tailcc/swifttailcc functions.

Pseudo-expansion's handling of tail-calls currently interacts badly
with shrink-wrapping when using callee-pop CCs on x86_64, resulting
in an unbalanced RSP adjustment on shrink-wrapped return paths.

Disable shrink-wrapping until that's resolved, if it can be.

See https://github.com/llvm/llvm-project/issues/109279

rdar://141673907
(cherry picked from commit 14065737a74eba248122bd9f2541b23432221048)

ahmedbougacha commented:
@swift-ci test

drexin reviewed (approved):
LGTM

jroelofs commented on llvm/test/CodeGen/X86/x86-tailcc-shrink-wrapping.ll:
```diff
@@ -0,0 +1,85 @@
+declare tailcc void @f2()
+declare tailcc void @f1(ptr, i64, ptr, i8, ptr, ptr, ptr, ptr, ptr, ptr)
+
+define tailcc void @test_shrink_wrap_swifttailcc(i64 %0) {
```
which thing is supposed to be `swifttailcc` re: the function's name? none of the declarations/definitions in the file have that calling convention.

ahmedbougacha replied:
Great eye, fixed!

ahmedbougacha committed:
[X86] Also test swifttailcc in shrink-wrapping disablement. NFC.

14065737a74e disabled shrink-wrapping for tailcc/swifttailcc functions.
But I copy-pasta'd the test into only testing tailcc.
Also test swifttailcc.

rdar://141673907
(cherry picked from commit e859ac81caf0d9f4cc264408b88cd5e5daae9ff8)

al45tair reviewed (approved):
:-) Thank-you Ahmed.

drexin commented:
@swift-ci test

porglezomp commented:
@swift-ci test

ahmedbougacha commented:
@swift-ci test Windows platform
